core/CHANGELOG.md

```diff
@@ -8,6 +8,7 @@ _YYYY-MM-DD_
 
 * Dependency loop error now lists all test cases that formed a cycle
 * `foldGroup` now takes `[b]` instead of `b` as its last argument to allow for custom fold strategies. This is a backwards incompatible change, but you can get the old behavior by applying `mconcat`.
+* Dependencies can now be defined pattern-free with `sequentialTestGroup`. (#343)
```

Comment:
What's the interaction of `--pattern` and dependencies? Is it unchanged? I was under impression that now if a test matches `--pattern`, its prerequisites are also executed, even if they don't match.

Reply:
That's correct, but _only_ for dependencies specified using `sequentialTestGroup`. Patterns defined using patterns retain their old behavior. (The initial approach did tackle this, but its implementation complexity was deemed too high.)

I see I didn't add a test for this, so I'll leave this conversation open until I've done that.